### PR TITLE
opendatahub-io/vllm: fix ibm build

### PR DESCRIPTION
**Pull request by dtrifiro**
force build09 for ibm-dev and non smoketest jobs

### CODE REVIEW

File: ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm-dev.yaml [hunk 1]
```diff
@@ -64,6 +66,7 @@ tests:
     clone: true
     from: vllm-build-ibm-dev-cuda
 - as: ibm-dev-image-mirror
+  cluster: build09
```
Comment (joerunde) [inline ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm-dev.yaml:69]: Should this be added to the build for `main` as well?

Comment (NickLucche) [reply ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm-dev.yaml:69]: I am not sure why maybe daniele knows best, but main builds are already getting routed to 09 https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/opendatahub-io_vllm/171/pull-ci-opendatahub-io-vllm-main-cuda-pr-image-mirror/1839688325670113280. 05 should pick up the jobs requesting a gpu

**Review by terrytangyuan — APPROVED** [review]
/lgtm
/approve

**Comment by dtrifiro** [comment]
closing as outdated

**Pull request closed without merging**